src/OpenGL/Extensions/Silk.NET.OpenGL.Extensions.ImGui/ImGuiController.cs

```diff
@@ -234,211 +146,474 @@ public void Update(float deltaSeconds)
-            _gl.BlendFunc(BlendingFactor.SrcAlpha, BlendingFactor.OneMinusSrcAlpha);
-            _gl.Disable(EnableCap.CullFace);
-            _gl.Disable(EnableCap.DepthTest);
+            DestroyFontsTexture();
+        }
 
-            // Render command lists
-            int vtxOffset = 0;
-            int idxOffset = 0;
-            for (int n = 0; n < drawData.CmdListsCount; n++)
+        /// <summary>
+        /// Frees ImGui Font texture.
+        /// </summary>
+        private void DestroyFontsTexture()
+        {
+            if (_fontTexture != 0)
             {
-                ImDrawListPtr cmdList = drawData.CmdListsRange[n];
-                for (int cmdI = 0; cmdI < cmdList.CmdBuffer.Size; cmdI++)
-                {
-                    ImDrawCmdPtr pcmd = cmdList.CmdBuffer[cmdI];
-                    if (pcmd.UserCallback != IntPtr.Zero)
-                    {
-                        throw new NotImplementedException();
-                    }
-                    else
-                    {
-                        _gl.ActiveTexture(TextureUnit.Texture0);
-                        _gl.BindTexture(TextureTarget.Texture2D, (uint)pcmd.TextureId);
-                        _gl.CheckGlError("Texture");
+                var io = ImGuiNET.ImGui.GetIO();
+                _gl.DeleteTexture(_fontTexture);
+                io.Fonts.SetTexID((IntPtr) 0);
+                _fontTexture = 0;
+            }
+        }
 
-                        // We do _windowHeight - (int)clip.W instead of (int)clip.Y because gl has flipped Y when it comes to these coordinates
-                        var clip = pcmd.ClipRect;
-                        _gl.Scissor((int)clip.X, _windowHeight - (int)clip.W, (uint)(clip.Z - clip.X), (uint)(clip.W - clip.Y));
-                        _gl.CheckGlError("Scissor");
+        private bool CheckShader(uint handle, string desc)
+        {
+            _gl.GetShader(handle, GLEnum.CompileStatus, out int status);
 
-                        _gl.DrawElementsBaseVertex(PrimitiveType.Triangles, (uint)pcmd.ElemCount, DrawElementsType.UnsignedShort, (void*)(idxOffset * sizeof(ushort)), vtxOffset);
-                        _gl.CheckGlError("Draw");
-                    }
+            if (status == 0)
+            {
+                string info = _gl.GetShaderInfoLog(handle);
+                Debug.WriteLine($"GL.CompileShader for shader '{desc}' had info log:\n{info}");
+            }
 
-                    idxOffset += (int)pcmd.ElemCount;
-                }
-                vtxOffset += cmdList.VtxBuffer.Size;
+            return status == 1;
+        }
+
+        private bool CheckProgram(uint handle, string desc)
+        {
+            _gl.GetProgram(handle, GLEnum.LinkStatus, out int status);
+
+            if (status == 0)
+            {
+                string info = _gl.GetProgramInfoLog(handle);
+                Debug.WriteLine($"GL.LinkProgram had info log [{desc}]:\n{info}");
             }
```

Comment:
This branch doesn't do anything but an unnecessary native call on Release, which is the config what we ship in. Recommended removal.

Reply:
Switched back to Shader.cs